docs/development/design/cockroachdb.md

```diff
@@ -0,0 +1,22 @@
+# Support Cockroach DB
+
+> [Issue #598](https://github.com/redhat-et/issues/598)
+
+## Summary
+
+Support a DB technology that can reduce DB downtime of the service and allow the DB to horizontally scale.  
+
+## Proposal
+
+We should support Cockroach DB in addition to Postgres.  Cockroach shards and replicates over several work nodes allowing it to be horizontally scalable and have higher availability than traditional single node databases.  It also support doing zero downtime upgrades of the database.
+
+Cockroach is already highly compatible with Postgres, it uses the same client drivers and supports most of the same SQL dialect.
```

Comment:
There are some issues with `gorm` and cockroachdb right? We should document those here and we plan to work around them OR fix them.

Reply:
doen.